### PR TITLE
Support scan ASCA command (AST-42348, AST-69533)

### PR DESCRIPTION
**Pull request by AlvoBen**

### CODE REVIEW

**Comment by AlvoBen** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [38cadac9\-ef41\-42f5\-8245\-f86d2e62a1b0](https://eu.ast.checkmarx.net/projects/df2dd522-8671-4034-acc7-e9fdd567fb26/scans?id=38cadac9-ef41-42f5-8245-f86d2e62a1b0&branch=feature%2Fbenalvo%2Fadd-asca-functionality)
### No New Or Fixed Issues Found

File: src/main/java/com/checkmarx/ast/ScanResult/ScanDetail.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    String severity;
+    String fileName;
+    int line;
+    String problematicLine;
+    int length;
+    String remediationAdvise;
+    String description;
+
+    @JsonCreator
+    public ScanDetail(
```
Comment (OrShamirCM) [inline src/main/java/com/checkmarx/ast/ScanResult/ScanDetail.java:30]: Change to Asca or have Asca folder

File: src/main/java/com/checkmarx/ast/ScanResult/ScanResult.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import lombok.Value;
+
+import java.util.List;
+
+@Value
+@ToString(callSuper = true)
+@JsonDeserialize()
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ScanResult {
```
Comment (OrShamirCM) [inline src/main/java/com/checkmarx/ast/ScanResult/ScanResult.java:20]: Asca

File: src/main/java/com/checkmarx/ast/wrapper/CxWrapper.java [hunk 3]
```diff
@@ -336,7 +365,7 @@ public KicsRealtimeResults kicsRealtimeScan(@NonNull String fileSources, String
         List<String> arguments = new ArrayList<>();
         arguments.add(CxConstants.CMD_SCAN);
         arguments.add(CxConstants.SUB_CMD_KICS_REALTIME);
-        arguments.add(CxConstants.FILE_SOURCES);
+        arguments.add(CxConstants.FILE_SOURCE);
```
Comment (OrShamirCM) [inline src/main/java/com/checkmarx/ast/wrapper/CxWrapper.java:368]: change to file

File: src/test/java/com/checkmarx/ast/ScanTest.java [hunk 4]
```diff
@@ -19,6 +23,43 @@ void testScanShow() throws Exception {
...
+            Assertions.assertNotNull(scanDetail.getDescription(), "Description should not be null");
+        }
+    }
+
+
+    @Test
+    void testScanAsca_WhenFileWithoutVulnerabilitiesIsSent_ReturnSuccessfulResponseWithCorrectValues() throws Exception {
+        ScanResult scanResult = wrapper.ScanAsca("src/test/resources/csharp-no-vul.cs", true, null);
+        Assertions.assertNotNull(scanResult.getRequestId());
+        Assertions.assertTrue(scanResult.isStatus());
```
Comment (OrShamirCM) [inline src/test/java/com/checkmarx/ast/ScanTest.java:51]: assert scan destiles nulol

**Pull request merged**